cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchCristinPersonHandlerTest.java

```diff
@@ -184,6 +196,86 @@ void shouldReturnGatewayTimeoutWhenFetchFromUpstreamServerTimesOut() throws Exce
         assertThat(gatewayResponse.getBody(), containsString(ERROR_MESSAGE_GATEWAY_TIMEOUT));
     }
 
+    @Test
+    void shouldNotReturnEmploymentsWhenNotAuthorized() throws IOException, ApiGatewayException {
+        apiClient = spy(apiClient);
+        handler = new FetchCristinPersonHandler(apiClient, environment);
+        var actual = sendQuery(EMPTY_MAP, VALID_PATH_PARAM).getBodyObject(Person.class);
+
+        verify(apiClient, times(0)).fetchGetResultWithAuthentication(any());
+        assertThat(actual.getEmployments().size(), equalTo(0));
+    }
+
+    @Test
+    void shouldReturnEmploymentsWhenAuthorized() throws IOException, ApiGatewayException {
+        apiClient = spy(apiClient);
+        var cristinPersonUri = URI.create(EXPECTED_CRISTIN_URI_WITH_IDENTIFIER);
+        doReturn(getResponseWithoutEmployment()).when(apiClient).fetchGetResultWithAuthentication(cristinPersonUri);
+        var cristinPersonEmploymentsUri = URI.create(EXPECTED_CRISTIN_EMPLOYMENTS_URI_WITH_IDENTIFIER);
+        doReturn(getResponseWithOnlyEmployments()).when(apiClient)
+            .fetchGetResultWithAuthentication(cristinPersonEmploymentsUri);
+        handler = new FetchCristinPersonHandler(apiClient, environment);
+        var actual = sendAuthorizedQuery(EDIT_OWN_INSTITUTION_USERS).getBodyObject(Person.class);
+
+        assertThat(actual.getEmployments().size(), equalTo(EXPECTED_EMPLOYMENT_SIZE_WHEN_AUTHORIZED));
+    }
+
+    @Test
+    void shouldReturnOkEvenWhenEmploymentsAreEmpty() throws ApiGatewayException, IOException {
+        apiClient = spy(apiClient);
+        var cristinPersonUri = URI.create(EXPECTED_CRISTIN_URI_WITH_IDENTIFIER);
+        doReturn(getResponseWithoutEmployment()).when(apiClient).fetchGetResultWithAuthentication(cristinPersonUri);
+        var cristinPersonEmploymentsUri = URI.create(EXPECTED_CRISTIN_EMPLOYMENTS_URI_WITH_IDENTIFIER);
+        doReturn(new HttpResponseFaker(EMPTY_JSON_ARRAY, 200)).when(apiClient)
+            .fetchGetResultWithAuthentication(cristinPersonEmploymentsUri);
+        handler = new FetchCristinPersonHandler(apiClient, environment);
+        var actual = sendAuthorizedQuery(EDIT_OWN_INSTITUTION_USERS);
+        var employments = actual.getBodyObject(Person.class).getEmployments();
+
+        assertThat(actual.getStatusCode(), equalTo(HttpURLConnection.HTTP_OK));
+        assertThat(employments.size(), equalTo(0));
+    }
+
+    @Test
+    void shouldNotReturnEmploymentsWhenNotHavingRequiredAccessRight() throws IOException, ApiGatewayException {
```

Comment:
use the "setUpApiClientToRespondToPersonAndEmploymentRequestsWhenRequestIsAuthorizedAndPersonHasEmployment" method